helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -202,6 +211,30 @@ protected void handleEvent(NotificationContext event) {
     }
   }
 
+  class RefreshTask implements Runnable {
+    @Override
+    public void run() {
+      try {
+        long currentTime = System.currentTimeMillis();
+        long remainingTime = _lastEventTime + _periodicRefreshInterval - currentTime;
+        if (remainingTime <= 0) {
```

Comment:
What if there is already one periodic event in the queue? Shall we keep adding new ones?